Headers/DebugServer2/Target/ThreadBase.h

```diff
@@ -57,6 +57,9 @@ class ThreadBase {
                            Address const &address = Address()) = 0;
 
 public:
+  virtual ErrorCode beforeResume();
```

Comment:
This `beforeResume()` mechanism seems a bit fragile to me.
I'm thinking we should always call `beforeResume()` before resuming a thread (maybe we should call it from the `resume()` function itself, but I'm not sure that's the cleanest way of implementing it); and that way you'd have to check if the stop reason is `kReasonThreadEntry` in `beforeResume()` before enabling watchpoints.